subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/ShadowPluginSmokeTest.groovy

```diff
@@ -49,15 +49,19 @@ class ShadowPluginSmokeTest extends AbstractPluginValidatingSmokeTest {
             """.stripIndent()
 
         when:
-        def result = runner('shadowJar').build()
+        def result = runner('shadowJar')
+            .expectLegacyDeprecationWarning(BaseDeprecations.FILE_TREE_ELEMENT_GET_MODE_DEPRECATION)
```

Comment:
We are testing the latest version of shadow.
This is not a "legacy" deprecation warning.
We can't ship with such nagging that users can't do anything about.

What we should do is the following:
* remove the nagging for now
* open issues with third parties plugin to move to the new API once available
* open issues on our side to add the nagging once there are releases of fixed 3rd parties available with links to these issues, and schedule them for the next milestone for a revisit